### PR TITLE
chore: Fix extra space at the end of multi-line input widget #33099

### LINKED ISSUE #33099: [Bug]: Extra space at the end of multi-line input widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Extra space equal to a line space appears at the end of the input box in a multi-line setting. The bottom gap/space needs to be lesser than line height.

image

### Steps To Reproduce

1. DnD an input widget and change to multi-line type
2. Drop a large text in preview mode and observe the extra line space at the end of the input area

### Public Sample App

_No response_

### Environment

Deploy Preview

### Severity

Medium (Frustrating UX)

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #33099

/ok-to-test tags="@tag.Anvil"
> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9318021083>
> Commit: 5f1ea63400f06fa9d22e0aa41c81e44fdf7b1deb
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9318021083&attempt=1" target="_blank">Click here!</a>

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/TextArea/src/TextArea.tsx [hunk 1]
```diff
@@ -48,9 +48,17 @@ function TextArea(props: TextAreaProps, ref: TextAreaRef) {
...
+      const paddingTop = parseFloat(computedStyle.paddingTop);
+      const paddingBottom = parseFloat(computedStyle.paddingBottom);
       input.style.height = `${
-        input.scrollHeight + (input.offsetHeight - input.clientHeight)
+        // subtract comptued padding and border to get the actual content height
+        input.scrollHeight -
+        paddingTop -
+        paddingBottom +
+        // Also, adding 1px to fix a bug in browser where there is a scrolllbar on certain heights
+        1
```
Comment (jsartisan) [inline app/client/packages/design-system/headless/src/components/TextArea/src/TextArea.tsx:61]: without +1, I see this bug

https://github.com/appsmithorg/appsmith/assets/6636360/754cf52c-d5ab-4505-9298-0e3b4b711e81

@KelvinOm

Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/TextArea/src/TextArea.tsx:61]: @jsartisan This works without adding `1`. Please check it. It looks like the calculations were simply wrong before.

Comment (jsartisan) [reply app/client/packages/design-system/headless/src/components/TextArea/src/TextArea.tsx:61]: For me, without 1, it does not work. 
I have these settings on my mac.
CleanShot 2024-06-03 at 15 23 18@2x

Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/TextArea/src/TextArea.tsx:61]: I see.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**